File: 028/028.php
<?php

function a($n) {
    return floor($n * ($n + 2) / 4) + floor(($n % 4) / 3) + 1;
}

$n = 1001;
$sum = 0;
for ($i = 1; a($i) <= $n*$n; $i++)
    $sum += a($i);
print($sum . "\n");

File: solutions/028/028.cc
#include <iostream>

int a(int n) {
    return n*(n+2)/4 + (n%4)/3 + 1;
}

int main() {
    int n = 1001;
    int sum = 0;
    for (int i = 1; a(i) <= n*n; i++)
        sum += a(i);
    std::cout << sum << std::endl;
}
